### ISSUE TITLE
Unclear listen setting

### ISSUE THREAD

**Issue by asergunovkl** [genotrance/px#200]
Hi. I'm using px on Windows and trying to make it less invasive as possible. So I was trying to limit interfaces it listens with `listen=` config. The plan was to listen only local loop, hyper-v and virtual box adapters. But for some reason it shows me in log also connections from LAN adapter.

**Comment by genotrance** [comment]
Can you please share your configuration?

**Comment by asergunovkl** [comment]
Sure.
```
# listening only local loop and WSL device
listen = 127.0.0.1,172.21.48.1
# Allow answering to WSL machines
gateway = 1
# to make sure it will not interfere other settings
hostonly = 0 
# this one I'm using as workaround to make sure only these clients will be served
allow = 127.0.0.1/8,172.21.48.1/20 
```

**Comment by genotrance** [comment]
`listen` only supports one interface at this time. There's issue #195 to support multiple interfaces that is on the roadmap.

`gateway` simply overrides `listen`. From the docs:
```
  Allow remote machines to use proxy. 0 or 1, default: 0
    Overrides 'listen' and binds to all interfaces
```

That's why Px is listening on everything. `allow` feature exists to filter only the clients you trust. So you are using those flags as expected.

**Comment by asergunovkl** [comment]
Oh I see. This override was not clear for me. I would be great if px communicated that clearly in debug mode. Currently it shows `listen=<empty>` and it's not clear what caused that from the output. But config I had was already modified so I didn't have this hint.

**Comment by genotrance** [comment]
Makes sense - I'll add some logging to provide hints.

**Comment by asergunovkl** [comment]
Thanks a lot!

**Comment by genotrance** [comment]
This is fixed in v0.9.0 still in development - see [branch](https://github.com/genotrance/px/tree/v090).

**Comment by asergunovkl** [comment]
That's awesome! Thanks a lot!
